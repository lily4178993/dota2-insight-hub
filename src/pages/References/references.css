.references-container {
  background: #d2e7f1;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 0 10px var(--color-black-2);
}

.references-container * {
  color: black;
}

.references-list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.reference-item {
  margin-bottom: 20px;
  padding: 20px;
  background: #fff;
  border: 1px solid #e0e0e0;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  transition: transform 0.2s ease-in-out;
}

.reference-item:hover {
  transform: translateY(-5px);
}

.references-container h2 {
  font-size: 1.8em;
  color: #333;
  margin-bottom: 20px;
}

.references-container h3 {
  font-size: 1.4em;
  color: #555;
  margin: 0 0 10px;
}

.references-container p {
  font-size: 1em;
  color: #777;
  margin: 0 0 15px;
}

.references-container a {
  font-size: 1em;
  color: #ffb6eb;
  text-decoration: none;
  transition: color 0.2s;
}

.references-container a:hover {
  color: #fb63d3;
}
