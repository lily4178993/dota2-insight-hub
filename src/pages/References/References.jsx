import React from 'react';
import './references.css';

function References() {
  const referencesList = [
    {
      name: 'Dota 2 API',
      description: 'Official API for Dota 2 game data.',
      link: 'https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/Scripting/API',
    },
    {
      name: 'React.js',
      description: 'JavaScript library for building user interfaces.',
      link: 'https://reactjs.org',
    },
    {
      name: 'Redux Toolkit',
      description: 'State management library for React applications.',
      link: 'https://redux-toolkit.js.org',
    },
    {
      name: 'OpenAI GPT-3.5',
      description: 'Natural language processing AI model developed by OpenAI.',
      link: 'https://openai.com/gpt-3',
    },
    {
      name: 'Codegrid React Navigation Menu Tutorial',
      description: 'Tutorial on implementing navigation menu and styles in React.',
      link: 'https://youtu.be/YAjENHeD2TM',
    },
    {
      name: 'Icons8',
      description: 'Provider of the application favicon.',
      link: 'https://icons8.com',
    },
    {
      name: 'Nelson Sakwa',
      description: 'Provider of the application original design idea.',
      link: 'https://www.behance.net/sakwadesignstudio',
    },
  ];

  return (
    <div className="references-container">
      <h2>References</h2>
      <ul className="references-list">
        {referencesList.map((reference, index) => (
          // eslint-disable-next-line react/no-array-index-key
          <li key={index} className="reference-item">
            <h3>{reference.name}</h3>
            <p>{reference.description}</p>
            <a href={reference.link} target="_blank" rel="noopener noreferrer">
              {reference.link}
            </a>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default References;
