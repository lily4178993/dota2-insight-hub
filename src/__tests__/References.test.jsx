import React from 'react';
import { render } from '@testing-library/react';
import References from '../pages/References/References';

test('References component matches snapshot', () => {
  const { asFragment } = render(<References />);
  expect(asFragment()).toMatchSnapshot();
});
