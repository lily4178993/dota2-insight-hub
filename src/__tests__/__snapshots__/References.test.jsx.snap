// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`References component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="references-container"
  >
    <h2>
      References
    </h2>
    <ul
      class="references-list"
    >
      <li
        class="reference-item"
      >
        <h3>
          Dota 2 API
        </h3>
        <p>
          Official API for Dota 2 game data.
        </p>
        <a
          href="https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/Scripting/API"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/Scripting/API
        </a>
      </li>
      <li
        class="reference-item"
      >
        <h3>
          React.js
        </h3>
        <p>
          JavaScript library for building user interfaces.
        </p>
        <a
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://reactjs.org
        </a>
      </li>
      <li
        class="reference-item"
      >
        <h3>
          Redux Toolkit
        </h3>
        <p>
          State management library for React applications.
        </p>
        <a
          href="https://redux-toolkit.js.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://redux-toolkit.js.org
        </a>
      </li>
      <li
        class="reference-item"
      >
        <h3>
          OpenAI GPT-3.5
        </h3>
        <p>
          Natural language processing AI model developed by OpenAI.
        </p>
        <a
          href="https://openai.com/gpt-3"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://openai.com/gpt-3
        </a>
      </li>
      <li
        class="reference-item"
      >
        <h3>
          Codegrid React Navigation Menu Tutorial
        </h3>
        <p>
          Tutorial on implementing navigation menu and styles in React.
        </p>
        <a
          href="https://youtu.be/YAjENHeD2TM"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://youtu.be/YAjENHeD2TM
        </a>
      </li>
      <li
        class="reference-item"
      >
        <h3>
          Icons8
        </h3>
        <p>
          Provider of the application favicon.
        </p>
        <a
          href="https://icons8.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://icons8.com
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
